gammapy/maps/wcs.py

```diff
@@ -551,6 +556,47 @@ def to_cube(self, axes):
         axes = copy.deepcopy(self.axes) + axes
         return self.__class__(self._wcs.deepcopy(), npix, cdelt=cdelt, axes=axes)
 
+    def pad(self, pad_width):
+        if np.isscalar(pad_width):
+            pad_width = (pad_width, pad_width)
+        npix = (self.npix[0] + 2 * pad_width[0],
+                self.npix[1] + 2 * pad_width[1])
+        wcs = self._wcs.deepcopy()
+        wcs.wcs.crpix += np.array(pad_width)
+        return self.__class__(wcs, npix, cdelt=copy.deepcopy(self._cdelt),
+                              axes=copy.deepcopy(self.axes))
+
+    def crop(self, crop_width):
```

Comment:
Why is the crop limited to a box around the center of the WcsGeom?

Reply:
Here the intent was to reproduce the functionality of the corresponding `SkyImage` methods.  We could certainly have a discussion about extending the functionality.  For what you're suggesting one approach might be a new method `crop_to(geom)` which would create a new map from the all pixels contained within the input geometry.  Alternatively one could add an optional parameter to `crop` that allows you to shift the registration point of the box in X/Y.